### ISSUE TITLE
Logging response headers from New Relic endpoint to audit logs.

### ISSUE THREAD

**Issue by vuqtran88** [newrelic/newrelic-dotnet-agent#112]
### Feature Description
When we send data up to new relic, the response headers that we get back from New Relic have been useful for debugging issues like missing data due to 503s, it's useful to know if the issues were isolated to a particular geo / colo. Additionally, if the failure is cloudflare related, cloudflare is going to need a ray ID and request ID to debug these issues.

**Comment by nrcventura** [comment]
I don't think that audit logs is the right place for this information. I think adding them to the DEBUG level agent logs would be more appropriate, which is where we are already logging the response body.

**Comment by vuqtran88** [comment]
I have changed the title.

**Comment by angelatan2** [comment]
12/10/2021 The .NET agent may need to support this feature as it is in the x-agent spec.
